dbt/semver.py

```diff
@@ -1,8 +1,11 @@
 import re
+import logging
 
 from dbt.exceptions import VersionsNotCompatibleException
 import dbt.utils
 
+logger = logging.getLogger(__name__)
```

Comment:
We usually do:

```
from dbt.logger import GLOBAL_LOGGER as logger
```

Is there any difference to grabbing the logger this way?

Reply:
Ah I see. Well I have a few thoughts on this. To answer your question, there is a difference. Using `__name__` means the logger in this module is scoped to `dbt.semver`. So you could, for example, use this to filter or format logs for this module differently than everything else. Like

```python
logging.getLogger("dbt.semver").setLevel(logging.ERROR)
```

But it also helps identify where logs are coming from if you were to put `%(name)s` in the format of the log messages since the name of the module would be in the logs.

You might not care about this, but one other thing I noticed is `GLOBAL_LOGGER` does not have a name, it's just:

```
logger = logging.getLogger()
```

which is the root logger. I would at least recommend changing this to `logger = logging.getLogger("dbt")`. Doing so would at least make [this](https://github.com/fishtown-analytics/dbt/blob/development/dbt/logger.py#L9-L14) a lot simpler, since your logging config could just look like:

```python
root_logger = logging.getLogger()
root_logger.setLevel(logging.CRITICAL)
logger = logging.getLogger("dbt")
logger.setLevel(logging.DEBUG)
```

and then you can whitelist modules you want to be more chatty instead of blacklisting modules.

I just tested all this out to confirm it's correct and the other thing you'd need to remove is https://github.com/fishtown-analytics/dbt/blob/development/scripts/dbt#L7 -- I find the `basicConfig` function is usually not helpful for anything other than one-off / simple scripts.

Reply:
This is super helpful, I think we will do this! Thanks for the heads up!

Reply:
@b-ryan this worked super well, thanks so much for the tip :)